qemu/tests/qemu_guest_agent.py

```diff
@@ -1829,25 +1831,125 @@ def gagent_check_nonexistent_cmd(self, test, params, env):
             test.fail("Should return error info.")
 
     @error_context.context_aware
-    def gagent_check_with_migrate(self, test, params, env):
+    def gagent_check_umount_frozen(self, test, params, env):
         """
-        Migration test with guest agent service running.
+        Umount file system while fs freeze.
+
+        Steps:
+        1) boot guest with a new data storage.
+        2) format disk and mount it.
+        3) freeze fs.
+        4) umount fs/offline the volume in guest,
+           for rhel6 guest will umount fail.
+        5) thaw fs.
+        6) mount fs and online it again.
 
         :param test: kvm test object
         :param params: Dictionary with the test parameters
-        :param env: Dictionary with test environment.
+        :param env: Dictionary with test environment
         """
-        error_context.context("Migrate guest while guest agent service is"
-                              " running.", logging.info)
-        self.vm.monitor.migrate_set_speed(params.get("mig_speed", "1G"))
-        self.vm.migrate()
-        error_context.context("Recreate a QemuAgent object after vm"
-                              " migration.", logging.info)
-        self.gagent = None
-        args = [params.get("gagent_serial_type"), params.get("gagent_name")]
-        self.gagent_create(params, self.vm, *args)
-        error_context.context("Verify if guest agent works.", logging.info)
-        self.gagent_verify(self.params, self.vm)
+        def wrap_windows_cmd(cmd):
+            """
+            add header and footer for cmd in order to run it in diskpart tool.
+
+            :param cmd: cmd to be wrapped.
+            :return: wrapped cmd
+            """
+            disk = "disk_" + ''.join(random.sample(string.ascii_letters +
+                                                   string.digits, 4))
+            cmd_header = "echo list disk > " + disk
+            cmd_header += " && echo select disk %s >> " + disk
+            cmd_footer = " echo exit >> " + disk
+            cmd_footer += " && diskpart /s " + disk
+            cmd_footer += " && del /f " + disk
+            cmd += " >> " + disk
+            return " && ".join([cmd_header, cmd, cmd_footer])
+
+        session = self._get_session(params, self.vm)
+        image_size_stg0 = params["image_size_stg0"]
+
+        if params['os_variant'] == 'rhel8':
+            session.cmd("setenforce 0")
```

Comment:
You might need update this later when we have newest release. I will prefer something like:
```
setenforce = True
Windows, RHEL.6, RHEL.7:
    setenforce = False
```
Or if it does not affect old release test, just send the command for all linux guest.

Reply:
I delete this part,as there is a product bug of rhel8, these code are just used to debug.
Thanks.